DESCW-1134: Add backup cronjob to PMO

robrien-bcgov opened the pull request:
https://apps.itsm.gov.bc.ca/jira/browse/DESCW-1134

robrien-bcgov committed:
DESCW-1134: Add backup cronjob to PMO

https://apps.itsm.gov.bc.ca/jira/browse/DESCW-1134

robrien-bcgov committed:
descw-1134 - add backup cronjob, guessing some config values

NOW i'm stuck.

robrien-bcgov committed:
descw-1134 pmo backups (#2) NOW I'm stuck lol

robrien-bcgov committed:
fix typo

ShawnTurple commented on deployments/kustomize/overlays/openshift/images/base/backups/README.md:
```diff
@@ -0,0 +1,109 @@
+resources:
+  - github.com/bcgov/gdx-agreements-tracker/deployments/kustomize/base
+# Update to your license plate in the dev|test|prod namespace.
+namespace: acd38d-dev
```
in the actual gdx repo, you can't add any reference to namespaces. 
```suggestion
namespace: default
```

ShawnTurple commented on deployments/kustomize/overlays/openshift/images/base/backups/README.md:
```diff
@@ -0,0 +1,109 @@
+namespace: acd38d-dev
+images:
+  - name: backup-container-mariadb
+    newName: image-registry.openshift-image-registry.svc:5000/acd38d-tools/backup-container-mariadb
```
```suggestion
    newName: backup-container-postgres
```

ShawnTurple commented on deployments/kustomize/overlays/openshift/images/base/sidecar/deployment.yaml:
```diff
@@ -0,0 +1,27 @@
+apiVersion: apps/v1
```
no need for a sidecar with gdx agreements tracker

ShawnTurple commented on deployments/kustomize/overlays/openshift/images/base/backups/README.md:
```diff
@@ -0,0 +1,109 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - github.com/bcgov/gdx-agreements-tracker/deployments/kustomize/base
```
```suggestion
  - github.com/bcgov/gdx-agreements-tracker/deployments/kustomize/base/backups
```

ShawnTurple commented on deployments/kustomize/overlays/openshift/images/base/backups/README.md:
```diff
@@ -0,0 +1,109 @@
+# Update to your license plate in the dev|test|prod namespace.
+namespace: acd38d-dev
+images:
+  - name: backup-container-mariadb
```
```suggestion
  - name: backup-container-postgres
```

ShawnTurple commented on deployments/kustomize/overlays/openshift/images/base/backups/README.md:
```diff
@@ -0,0 +1,109 @@
+      - BACKUP_DIR=/backups/my-backup-folder/
+      - MYSQL_USER=db_user
+      # format backup.conf=mariadb={db service}:3306/{db name}
+      - backup.conf=mariadb=gdx-agreements-tracker-mariadb:3306/db_name
```
using postgres, not mariadb

robrien-bcgov committed:
Update deployments/kustomize/overlays/openshift/images/base/backups/README.md

Co-authored-by: Shawn <45861534+ShawnTurple@users.noreply.github.com>

robrien-bcgov committed:
Update deployments/kustomize/overlays/openshift/images/base/backups/README.md

Co-authored-by: Shawn <45861534+ShawnTurple@users.noreply.github.com>

robrien-bcgov committed:
Update deployments/kustomize/overlays/openshift/images/base/backups/README.md

Co-authored-by: Shawn <45861534+ShawnTurple@users.noreply.github.com>

robrien-bcgov committed:
Update deployments/kustomize/overlays/openshift/images/base/backups/README.md

Co-authored-by: Shawn <45861534+ShawnTurple@users.noreply.github.com>

robrien-bcgov committed:
remove sidecar from base image and update documentation

robrien-bcgov committed:
change structure of kustomization - still needs more work before i can apply it

robrien-bcgov committed:
descw-1134_pmo-backups move folder structure to match other deployments

robrien-bcgov committed:
doing experiments - changed the structure of files

* errors need to be fixed still:

```sh
oc apply -f deployments/kustomize/overlays/openshift/experimental/openshift/output.yaml
configmap/pmo-backup-config-8kff8k2mmg unchanged
secret/pmo-backup-secrets-97dk5ch26h unchanged
persistentvolumeclaim/pmo-backup-pvc unchanged
persistentvolumeclaim/pmo-backup-verification-pvc unchanged
The CronJob "pmo-backup-cron" is invalid:
* spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[0].name: Not found: "pmo-backup-pvc"
* spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[1].name: Not found: "pmo-backup-verification-pvc"
* spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[2].name: Not found: "pmo-backup-config-volume"
```
just some incorrect values

robrien-bcgov committed:
includes changes except for the experimental local testing yaml

ShawnTurple commented on deployments/kustomize/backups/kustomization.yaml:
```diff
@@ -0,0 +1,24 @@
+      - backup.conf=postgres=deployment:5432/postgres
+      - POSTGRES_DB=secret_postgres_db
+      - POSTGRES_USER=secret_postgres_user
+      - POSTGRES_REGULAR_USER=secret_postgres_regular_user
```
Should we just remove this, as it is not going to be used.

robrien-bcgov replied:
good point. I'll try to build/apply without it and see what happens

robrien-bcgov committed:
remove redundant kustomization stuff from backups

robrien-bcgov committed:
remove whitespace changes from extisting kustomization

robrien-bcgov committed:
add volume/claim to hold backups for inspection in postgresql deploy

robrien-bcgov committed:
correct kustomization for backups to allow for merge/replace on tenant repo

robrien-bcgov committed:
remove deployment changes and set reasonable defaults

robrien-bcgov committed:
linting

robrien-bcgov committed:
remove namePrefix from base kustomization

robrien-bcgov committed:
fix secrets name and remove nameprefix

ShawnTurple commented on deployments/kustomize/backups/kustomization.yaml:
```diff
@@ -0,0 +1,23 @@
+      - POSTGRES_USER=secret_postgres_user
+      - POSTGRES_REGULAR_USER=secret_postgres_regular_user
+      - POSTGRES_HOST=secret_postgres_host
+secretGenerator:
```
remove secretsgenerator as we don't need.

robrien-bcgov committed:
remove redundant stuff
